Python/sort/Parallel_MergeSort.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+""" Merge Sort Algorithm for Sort an array of random integers with parallel and sequential apporach is used."""
+
+import random
+import math
+import multiprocessing as mp
+
+""" helper method to merge two sorted subarrays
+    array[l..m] and array[m+1..r] into array """
+def merge(array, left, mid, right):
+    # copy data to temp subarrays to be merged
+    left_temp_arr = array[left:mid+1].copy()
+    right_temp_arr = array[mid+1:right+1].copy()
+    # initial indexes for left, right and merged subarrays
+    left_temp_index = 0
+    right_temp_index = 0
+    merge_index = left
+    # merge temp arrays into original
+    while (left_temp_index < (mid - left + 1) or right_temp_index < (right - mid)):
+        if (left_temp_index < (mid - left + 1) and right_temp_index < (right - mid)):
+            if (left_temp_arr[left_temp_index] <= right_temp_arr[right_temp_index]):
+                array[merge_index] = left_temp_arr[left_temp_index]
+                left_temp_index += 1
+            else:
+                array[merge_index] = right_temp_arr[right_temp_index]
+                right_temp_index += 1
+        elif (left_temp_index < (mid - left + 1)): # copy any remaining on left side
+            array[merge_index] = left_temp_arr[left_temp_index]
+            left_temp_index += 1
+        elif (right_temp_index < (right - mid)): # copy any remaining on right side
+            array[merge_index] = right_temp_arr[right_temp_index]
+            right_temp_index += 1
+        merge_index += 1
+
+""" sequential implementation of merge sort """
+def sequential_mergesort(array, *args):
+    if not args: # first call
+        sequential_mergesort(array, 0, len(array)-1)
+        return array # return the sorted array
+    else: # recursive call
+        left, right = args
+        if (left < right):
+            mid = (left + right) // 2 # find the middle point
+            sequential_mergesort(array, left, mid) # sort the left half
+            sequential_mergesort(array, mid+1, right) # sort the right half
+            merge(array, left, mid, right) # merge the two sorted halves	
+	
+""" parallel implementation of merge sort """
+def parallel_mergesort(array, *args):
+    if not args: # first call
+        shared_array = mp.RawArray('i', array)
+        parallel_mergesort(shared_array, 0, len(array)-1, 0)
+        array[:] = shared_array # insert result into original array
+        return array
+    else:
+        left, right, depth = args
+        if (left < right):
+            mid = (left + right) // 2

```

Comment:
@PinkyWalia please avoid this technique to find the mid element since this can cause an integer overflow.
use this int mid = first + ((last - first) / 2);